mlflow/store/artifact/cloud_artifact_repo.py

```diff
@@ -244,17 +248,33 @@ def _parallelized_download_from_cloud(self, file_size, remote_file_path, local_p
                 env=parallel_download_subproc_env,
                 headers=self._extract_headers_from_credentials(cloud_credential_info.headers),
             )
-
-            if failed_downloads:
+            num_retries = _MLFLOW_MPD_NUM_RETRIES.get()
+            interval = _MLFLOW_MPD_RETRY_INTERVAL_SECONDS.get()
+            failed_downloads = list(failed_downloads)
+            while failed_downloads and num_retries > 0:
+                self._refresh_credentials()
                 new_cloud_creds = self._get_read_credential_infos([remote_file_path])[0]
                 new_signed_uri = new_cloud_creds.signed_uri
                 new_headers = self._extract_headers_from_credentials(new_cloud_creds.headers)
-                download_chunk_retries(
-                    chunks=list(failed_downloads),
-                    headers=new_headers,
-                    http_uri=new_signed_uri,
-                    download_path=local_path,
-                )
+
+                futures = {self.chunk_thread_pool.submit(download_chunk, range_start=chunk.start, range_end=chunk.end, headers=new_headers, download_path=local_path, http_uri=new_signed_uri): chunk for chunk in failed_downloads}
```

Comment:
nit: the futures code here seems to be duplicated from `parallelized_download_file_using_http_uri `? if we could modify that func to re-use its logic for downloading failed chunks, that may look cleaner.